mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageZipper.java

```diff
@@ -0,0 +1,87 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.commons.compress.archivers.zip.ExtraFieldUtils;
+import org.apache.commons.compress.archivers.zip.ZipArchiveEntry;
+import org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream;
+import org.apache.commons.io.IOUtils;
+import org.apache.james.mailbox.backup.MessageIdExtraField;
+import org.apache.james.mailbox.backup.SizeExtraField;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.store.streaming.CountingInputStream;
+
+import com.github.fge.lambdas.Throwing;
+
+import reactor.core.publisher.Mono;
+
+class DeletedMessageZipper {
+
+    private final DeletedMessageVault vault;
+
+    DeletedMessageZipper(DeletedMessageVault vault) {
+        ExtraFieldUtils.register(MessageIdExtraField.class);
+        ExtraFieldUtils.register(SizeExtraField.class);
+        this.vault = vault;
+    }
+
+    OutputStream zip(Stream<DeletedMessage> deletedMessages) throws IOException {
+        ByteArrayOutputStream deletedMessagesStream = new ByteArrayOutputStream();
+        ZipArchiveOutputStream zipOS = new ZipArchiveOutputStream(deletedMessagesStream);
+
+        deletedMessages
+            .forEach(Throwing.<DeletedMessage>consumer(deletedMessage -> putMessageToEntry(zipOS, deletedMessage)).sneakyThrow());
```

Comment:
If you sneakyThrow, you should have the exception in the signature.

BTW is it wize to abort the entire ZIP if a single entry failed?